### PR TITLE
104 auto submit after x digits entered

### LINKED ISSUE #104: Auto-submit after X digits entered
add a setting to configure a autosubmit after X digits have been entered

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
Feature:
- auto submit otp length

Refactor:
- separate the CSS from templates
- separate the JS from templates
- UI

### CODE REVIEW

File: twofactor_privacyidea/appinfo/info.xml [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
-Thus you can assign different authentication objects to your users as second factor. Users
-can now authenticate against your ownCloud installation with e.g. Google Authenticator,
-the [privacyIDEA Authenticator](https://netknights.it/en/produkte/privacyidea-authenticator-app/)
-Yubikey, U2F tokens, key fob, OTP display cards and much more.
+	<summary>Most flexible enterprise level Multi-Factor Authentication</summary>
+	<description>
+		This is a plugin to forward the authentication requests to your own privacyIDEA Authentication System.
+		[Your own privacyIDEA Authentication System](https://netknights.it/en/produkte/privacyidea/) can run on
+		premise centrally in your network under your control.
+		The privacyIDEA can manage many different kind of authentication devices. Thus, you can assign different
```
Comment (nilsbehlen) [inline twofactor_privacyidea/appinfo/info.xml]: just privacyIDEA

File: twofactor_privacyidea/css/pi-form-template.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
+.pi_images
```
Comment (nilsbehlen) [inline twofactor_privacyidea/css/pi-form-template.css:1]: this name uses _ while the other use camelCase. please use consistent notation

File: twofactor_privacyidea/l10n/de.json [hunk 3]
```diff
@@ -33,13 +33,15 @@
     "\n                    Verify the SSL certificate.\n                " : "\nDas SSL Zertifikat überprüfen.",
     "\n                        Do not uncheck this in productive environments!\n                    " : "\nBitte deaktiviere dies nicht in einer produktiven Umgebung!",
     "Ignore the system wide proxy settings and send authentication requests to privacyIDEA directly." : "Die systemweiten Proxy-Einstellungen ignorieren und den HTTP-Request für die Authentifizierung direkt an privacyIDEA senden.",
-    "Test authentication by supplying username and password that are checked against privacyIDEA:" : "Sie können die Konfiguration testen, indem Sie einen Benutzernamen und das Einmal-Passwort, das an privacyIDEA gesendet werden soll, hier eingeben:",
+    "Test the authentication by supplying username and password \n            that should be checked against the privacyIDEA:" : "Sie können die Konfiguration testen, indem Sie einen Benutzernamen und das Einmal-Passwort, das an privacyIDEA gesendet werden soll, hier eingeben:",
```
Comment (nilsbehlen) [inline twofactor_privacyidea/l10n/de.json]: both "the" are not needed

File: twofactor_privacyidea/templates/settings-admin.php [hunk 4]
```diff
@@ -1,162 +1,197 @@
...
+    </h2>
     <span id="pi_settings_msg" class="msg"></span>
     <p>
         <em>
-            <?php p($l->t('
-                In a second step of authentication the user is asked to provide a one
-                time password. The users devices are managed in privacyIDEA. The
-                authentication request is forwarded to privacyIDEA.
-            ')); ?>
+            <?php p($l->t('In a second step of authentication the user is asked to provide a one time password.
```
Comment (nilsbehlen) [inline twofactor_privacyidea/templates/settings-admin.php]: , after authentication

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: twofactor_privacyidea/appinfo/info.xml (hunk 1 above)
Comment (lukasmatusiewicz) [reply twofactor_privacyidea/appinfo/info.xml]: [c0c8eec](https://github.com/privacyidea/privacyidea-owncloud-app/pull/107/commits/c0c8eec64067356d0dda825987abc3ccf818fd92)

File: twofactor_privacyidea/l10n/de.json (hunk 3 above)
Comment (lukasmatusiewicz) [reply twofactor_privacyidea/l10n/de.json]: [f8aa402](https://github.com/privacyidea/privacyidea-owncloud-app/pull/107/commits/f8aa40280fe1e679b6c57aa1bcf92e4bb38738af)

File: twofactor_privacyidea/templates/settings-admin.php (hunk 4 above)
Comment (lukasmatusiewicz) [reply twofactor_privacyidea/templates/settings-admin.php]: [5f03ff5](https://github.com/privacyidea/privacyidea-owncloud-app/pull/107/commits/5f03ff573651aa5aa04f6ffb66203a1302a0945d)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**